src/bosh-dns/dns/server/handlers/failover_recursor_pool.go

```diff
@@ -89,26 +112,45 @@ func newSmartFailoverRecursorPool(recursors []string, logger logger.Logger) Recu
 		preferredRecursorIndex: 0,
 		logger:                 logger,
 		logTag:                 logTag,
+		recursorRetrySettings:  recursorSettings,
 	}
 }
 
 func (q *serialFailoverRecursorPool) PerformStrategically(work func(string) error) error {
 	for _, r := range q.recursors {
-		if err := work(r); err == nil {
+		if err := performWithRetryLogic(work, r, q.recursorRetrySettings.retryCount, q.logTag, q.logger); err == nil {
 			return nil
 		}
 	}
-
 	return ErrNoRecursorResponse
 }
 
+func performWithRetryLogic(work func(string) error, recursor string, retryCount int, logTag string, log logger.Logger) (err error) {
+	for ret := -1; ret < retryCount; ret++ {
+		err = work(recursor)
+		if err == nil {
+			return err
+		}
+		if _, ok := err.(net.Error); !ok {
+			log.Debug(logTag, fmt.Sprintf("dns request error %v no retry - %v\n", err, recursor))
```

Comment:
I think this is basically already logged when we switch recursors?

Reply:
You log it at the the worker function:
https://github.com/cloudfoundry/bosh-dns-release/blob/1e5250dd8bb790aede2a6c3d2f3999c573fc5d2d/src/bosh-dns/dns/server/handlers/forward_handler.go#L66

So I think it's fine to delete that line that we don't log to much